mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -11,29 +12,91 @@
 import math
 import itertools
 from datetime import date
+from pathlib import Path
+import glob
+import argparse
 
 import numpy as np
 import pandas as pd
 import re
 
 # global variables.
-MLIR_BIN_DIR = './bin'
-LLVM_BIN_DIR = './external/llvm-project/llvm/bin'
-MIOPEN_GEN = 'miopen-gen'
-MLIR_MIOPEN_DRIVER = 'mlir-miopen-driver'
-MLIR_ROCM_RUNNER = 'mlir-rocm-runner'
-MLIR_ROCM_RUNNER_ARGS = ['--shared-libs=./external/llvm-project/llvm/lib/libmlir_rocm_runtime.so,./lib/libconv-validation-wrappers.so,./external/llvm-project/llvm/lib/libmlir_runner_utils.so', '--entry-point-result=void']
 ROCPROF = '/opt/rocm/bin/rocprof'
-MIOPEN_DRIVER = '../MIOpen/build/bin/MIOpenDriver'
 BENCHMARKING_RESULT_FILE_NAME = 'results.stats.csv'
-CONFIGURATION_FILE_NAME ='../mlir/utils/jenkins/miopen-tests/resnet50-miopen-configs'
-
 DIRECTIONS = ['-F 1', '-F 2', '-F 4']
 DATA_TYPES = ['conv', 'convfp16', 'convint8']
 LAYOUTS = ['NHWC', 'NCHW']
+
 # Compiled regexp object used for extracting elapsed time from MIOpenDriver's output
 ELAPSED_TIME_RE = re.compile(r"Elapsed: (.*)ms")
 
+class Paths(NamedTuple):
+    """This structure is used to hold paths needed to perform the tests"""
+    miopen_gen_path : str
+    mlir_miopen_driver_path : str
+    miopen_driver_path : str
+    rocm_runner_path : str
+    libmlir_rocm_runtime_path : str
+    libconv_validation_wrappers_path : str
+    libmlir_runtime_utils_path : str
+    configuration_file_path : str
+
+
+def find_mlir_build_dir() -> str:
```

Comment:
Types aside, I'd like to see this first check for `./bin/miopen-gen` and `../../../build/bin/miopen-gen`. A full autodetector will complain about my home directory, which has both `~/llvm-project-mlir/` and `~/llvm-project-mlir-upstream/`

Reply:
Will the upstream one has miopen-opt as well ? 

Or

what is special about miopen-gen ?

Reply:
Sorry, `llvm-project-mlir-upstream` is a second clone of our repository I use to have a clean copy of the `miopen-dialect` branch to make testing easier.

And I'd just misspoken about which executable you were looking for, none of them matter

Reply:
Any suggestion to differentiate between those ?
or 
do we want to ? because the alternative is to provide a path to build dir

Reply:
Start from the path where the script is located. From there, test if there's a `./miopen-[something]`. If not, check to see if you're running from source by testing for a `../../../build/bin/miopen-[something]`. And then, as a final fallback, either tell the user to pass a path or do the search you're currently doing, either's probably fine.

Similarly, to get the MIOpen path (which you probably only want to test for if the user's test scheme asked for MIOpen runs), check for `[script directory]/../../MIOpen` and `[script directory]/../../../MIOpen/` to cover the built and source case, respectively.

Reply:
Makes sense! I ll change this tommorow when Im bored trying to figuring out other things.

Reply:
Done (I think)